parser/prism/Translator.cc

```diff
@@ -1866,7 +1866,18 @@ template <typename PrismNode> std::unique_ptr<parser::Mlhs> Translator::translat
                 auto location = translateLoc(splatNode->base.location);
                 auto expr = translate(splatNode->expression);
 
-                sorbetLhs.emplace_back(make_unique<parser::SplatLhs>(location, move(expr)));
+                std::unique_ptr<parser::Node> sorbetPlatNode;
+
+                auto argNode = dynamic_cast<parser::Arg *>(expr.get());
```

Comment:
Out of curiosity, what is the type of `expr` here? I'm wondering why we should `dynamic_cast` to a Sorbet parser type here rather than using some information in `splatNode->expression` and then translating directly into a `parser::Arg`?